logstash-core/src/main/java/org/logstash/instrument/witness/PluginsWitness.java

```diff
@@ -20,10 +30,12 @@ public PluginsWitness() {
         this.inputs = new ConcurrentHashMap<>();
         this.outputs = new ConcurrentHashMap<>();
         this.filters = new ConcurrentHashMap<>();
-     }
+        this.codecs = new ConcurrentHashMap<>();
```

Comment:
@jakelandis here it's not so much about throughput but more about saving memory. Default concurrency factor of `16` requires 16x as much memory compared to just putting a `1` for it (especially for cold path code). Hard to say how much it will matter here, but why pollute the heap when we don't have to :D